tower_cli/api.py

```diff
@@ -262,9 +266,27 @@ def test_mode(self):
                                      password='This is the best wine.',
                                      verbose=False, format='json'):
             adapters = copy.copy(self.adapters)
-            faux_adapter = FauxAdapter(
-                url_pattern=self.prefix.rstrip('/') + '%s',
-            )
+
+            class VersionedFauxAdapter(FauxAdapter):
+                # An adapter that doesn't track version detection requests
+                # (those sent to /api/
+                def send(self, request, stream=False, **kwargs):
+                    ret = super(VersionedFauxAdapter, self).send(request, stream, **kwargs)
```

Comment:
the FauxAdapter tracks a list of requests made to it, and we have *hundreds* of tests that make assertions about the tests contents - and reference them by index.  So we'll have tests that do things like:

The *first* request is to `/api/v2/xyz/`, the *second* request is a `POST` at ...

Because every API request now is preceded immediately by a `GET /api/` to determine the `current_version` of the API, _all_ of these tests are now broken.  So this change makes it so that the version determination requests are not recorded in `FauxAdapter.requests` (so our tests don't have to all be changed).